gravitee-exchange-connector/gravitee-exchange-connector-websocket/src/main/java/io/gravitee/exchange/connector/websocket/client/WebSocketEndpoint.java

```diff
@@ -30,16 +33,34 @@ public class WebSocketEndpoint {
     private static final int DEFAULT_HTTP_PORT = 80;
     private static final int DEFAULT_HTTPS_PORT = 443;
 
+    private final URL url;
     private final URI uri;
 
-    @Builder
-    public WebSocketEndpoint(final String url) {
-        this.uri = URI.create(url);
+    private WebSocketEndpoint(final URL url, final URI uri) {
+        this.url = url;
+        this.uri = uri;
+    }
+
+    /**
+     * Creates a new WebSocketEndpoint instance using the given URL.
+     *
+     * @param value the URL to be used for creating the WebSocketEndpoint
+     * @return an Optional containing the WebSocketEndpoint instance if the URL is valid, otherwise an empty Optional
+     */
+    public static Optional<WebSocketEndpoint> newEndpoint(String value) {
+        try {
+            URL url = new URL(value);
+            URI uri = url.toURI();
+
+            return Optional.of(new WebSocketEndpoint(url, uri));
+        } catch (MalformedURLException | URISyntaxException e) {
+            return Optional.empty();
```

Comment:
Should'nt we add a warn log to explicitly explain why there is no endpoint even if the configuration contains some?